### PR TITLE
EXSWCPHIPT-121 - hipGetDeviceFlags and hipSetDeviceFlags tests

### PR DESCRIPTION
**Pull request by FMarno**
stacked on #2664

### CODE REVIEW

File: tests/catch/unit/device/hipGetSetDeviceFlags.cc [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+/**
+ * Conformance test for checking functionality of
+ * hipError_t hipGetDeviceFlags(unsigned int* flags);
+ * hipError_t hipSetDeviceFlags(unsigned flags);
+ *
+ *
+ * hipGetDeviceFlags and hipSetDeviceFlags tests.
+ * Scenario1: Validates if hipGetDeviceFlags returns hipErrorInvalidValue for flags = nullptr.
+ * Scenario2: Validates if hipSetDeviceFlags returns hipErrorInvalidValue for invalid flags.
+ * Scenario3: Validates if flags returned by hipGetDeviceFlags is valid.
```
Comment (alistair-low) [inline tests/catch/unit/device/hipGetSetDeviceFlags.cc]: "is" -> "are"?

File: tests/catch/unit/device/hipGetSetDeviceFlags.cc [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+ * Scenario4: Validates that flags set with hipGetDeviceFlags can be retrieved with
```
Comment (alistair-low) [inline tests/catch/unit/device/hipGetSetDeviceFlags.cc]: "set with hipSetDeviceFlags"
Same typo in Scenario 5

**Review by alistair-low — COMMENTED** [review]
Looks good to me! I left some very minor comments about comments.

File: tests/catch/unit/device/hipGetSetDeviceFlags.cc (hunk 1 above)
Comment (FMarno) [reply tests/catch/unit/device/hipGetSetDeviceFlags.cc]: resolved.

File: tests/catch/unit/device/hipGetSetDeviceFlags.cc (hunk 2 above)
Comment (FMarno) [reply tests/catch/unit/device/hipGetSetDeviceFlags.cc]: changed both

**Review by chrispaquot — APPROVED** [review]

**Pull request merged**